airflow/jobs/scheduler_job.py

```diff
@@ -1509,9 +1512,10 @@ def _process_executor_events(self, simple_dag_bag, session=None):
         filter_for_tis = TI.filter_for_tis(tis_with_right_state)
         tis = session.query(TI).filter(filter_for_tis).all()
         for ti in tis:
-            key = ti.key
-            dag_id, task_id, execution_date, try_number = key
-            state, info = event_buffer.pop(key)
+            # Recreate ti_key (dag_id, task_id, execution_date, try_number) using in-memory try_number
+            buffer_key = (*ti.key[:-1], ti_key_to_try_number_map[ti.key[:-1]])
```

Comment:
I would prefer to keep readability and unpack all values here, and then create a new key.
```
dag_id, task_id, execution_date, _ = ti.key
new_try_number = ti_key_to_try_number_map[(dag_id, task_id, execution_date)])
buffer_key = dag_id, task_id, execution_date, new_try_number
```
Slicing of the tuples is hard to read. Slicing of tuples is useful if you have a set of homogeneous elements.

Reply:
Yes, I did this at first but then wanted to shorten it